pkg/k8s/k8siface.go

```diff
@@ -70,13 +72,14 @@ func (k *k8sIface) ListGatewayNodes() (*v1.NodeList, error) {
 }
 
 func (k *k8sIface) updateLabel(nodeName string, mutate func(existing *v1.Node)) error {
-	// nolint:wrapcheck // Let the caller wrap these errors.
 	client := &resource.InterfaceFuncs{
 		GetFunc: func(ctx context.Context, name string, options metav1.GetOptions) (runtime.Object, error) {
-			return k.clientSet.CoreV1().Nodes().Get(ctx, name, options)
+			node, err := k.clientSet.CoreV1().Nodes().Get(ctx, name, options)
+			return node, errors.Wrap(err, fmt.Sprintf("error getting node %q", name))
 		},
 		UpdateFunc: func(ctx context.Context, obj runtime.Object, options metav1.UpdateOptions) (runtime.Object, error) {
-			return k.clientSet.CoreV1().Nodes().Update(ctx, obj.(*v1.Node), options)
+			node, err := k.clientSet.CoreV1().Nodes().Update(ctx, obj.(*v1.Node), options)
+			return node, errors.Wrap(err, fmt.Sprintf("error updating node %q ", obj.(*v1.Node).Name))
```

Comment:
As mentioned before, these are pass-throughs so we don't need to wrap - let the caller do it.

Reply:
Linter was throwing an error if not wrapped

pkg/k8s/k8siface.go:81:11: error returned from interface method should be wrapped: sig: func (k8s.io/client-go/kubernetes/typed/core/v1.NodeInterface).Update(ctx context.Context, node *k8s.io/api/core/v1.Node, opts k8s.io/apimachinery/pkg/apis/meta/v1.UpdateOptions) (*k8s.io/api/core/v1.Node, error) (wrapcheck)
			return k.clientSet.CoreV1().Nodes().Update(ctx, obj.(*v1.Node), options)

Reply:
Right - that's why I previously had "_// nolint:wrapcheck // Let the caller wrap these errors._" on L73.

Reply:
oh ok i got it now, added the comment .